lib/fluent/plugin/out_azurestorage_gen2.rb

```diff
@@ -275,6 +276,7 @@ def acquire_access_token_msi
             params = { :"api-version" => ACCESS_TOKEN_API_VERSION, :resource => "#{@url_storage_resource}" }
             unless @azure_instance_msi.nil?
                 params[:msi_res_id] = @azure_instance_msi
+                params[:client_id] = @azure_client_id
```

Comment:
thanks for the contribution.
can you set this param the same way as the azure_instance_msi ? (put inside `unless @ azure_client_id.nil?`) just in order to not any break functionality if i will use a new upgraded version. (so it would be optional)
I think (based on MSFT docs) the client_id is required if your VM has multiple user-assigned managed identities, could be that fits on your use case

Reply:
whatever, i merging this as is and i will do a follow up change